utils/zerovec/src/map/vecs.rs

```diff
@@ -44,27 +73,93 @@ pub trait ZeroVecLike<'a, T: ?Sized> {
     fn clear(&mut self);
     /// Reserve space for `addl` additional elements
     fn reserve(&mut self, addl: usize);
-    /// Check if this vector is in ascending order according to `T`s `Ord` impl
-    fn is_ascending(&self) -> bool;
-    /// Check if this vector is empty
-    fn is_empty(&self) -> bool {
-        self.len() == 0
-    }
+    /// Construct a borrowed version from this
+    ///
+    /// Note: This really should be `&'b self -> Self::BorrowedVersion<'b>`
+    /// but doing that requires complicated `for<'b>` code that will likely trigger
+    /// compiler bugs. Instead, we hope that `self` is covariant so this cast will
+    /// just work in the implementation. Basically, we rely on the compiler
+    /// casting `&'b Self<'a>` to `&'b Self<'b>` when this gets called, which works
+    /// out for `ZeroVec` and `VarZeroVec` containers just fine.
+    fn as_borrowed(&'a self) -> Self::BorrowedVersion;
```

Comment:
Naming suggestions:

- `.as()`
- `.get()`
- `.ref()`
- ... ?

As suggested above, I think we should reserve the word "borrowed" for the actually fully borrowed type.  In this case, we're converting from the general mutable type to the type with common methods that could be either mutable or borrowed.

Reply:
No, BorrowedVersion is always the fully borrowed type

Reply:
The client of this API doesn't care about the ownership model of the return type.  They are just required to call this method to get to the reader methods like `.get()`.  If the client cares about the ownership model, they call `.as_maybe_borrowed()`.

Reply:
discussed and converted `maybe_as_borrowed()` -> `as_borrowed_inner()`